pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java

```diff
@@ -91,33 +92,51 @@ public SegmentProcessorFramework(List<RecordReader> recordReaders, SegmentProces
    */
   public List<File> process()
       throws Exception {
+    try {
+      return doProcess();
+    } catch (Exception e) {
+      if (_partitionToFileManagerMap != null) {
```

Comment:
Should we clean up _segmentsOutputDir as well ?

Reply:
good point. think we should, as that's resources created by this processor.